### ISSUE TITLE
Provide deterministic output of linter warnings

### ISSUE THREAD

**Issue by invidian** [golangci/golangci-lint#1513]
**Is your feature request related to a problem? Please describe.**

Currently output from running the linter is sorted randomly:
```
1 ✗ (533ms) 18:02:25 invidian@dellxps15mateusz ~/repos/kinvolk/flatcar-linux-update-operator (invidian/more-cleanups)*$ golangci-lint run ./... | sha256sum
ce6aef0b8ed653437b2ea9b089508d17f5d9da4eb88e2cd609f008db1fb10d1d  -
0 ✓ (512ms) 18:04:04 invidian@dellxps15mateusz ~/repos/kinvolk/flatcar-linux-update-operator (invidian/more-cleanups)*$ golangci-lint run ./... | sha256sum
4514787c1974b1a875005d03b05538bbec563d4280208a5a1c5745bb4d997861  -
0 ✓ (590ms) 18:04:06 invidian@dellxps15mateusz ~/repos/kinvolk/flatcar-linux-update-operator (invidian/more-cleanups)*$ golangci-lint run ./... | sha256sum
92df4334649dee39870b22b683c9f8abec00307806def963c10668b5c48fb014  -
```

This makes it difficult to fix one issue, then re-run the linter as all warnings change the order.

**Describe the solution you'd like**

Output from `golangci-lint run ./...` should be the same on every invocation if source code does not change.

**Comment by bombsimon** [comment]
The linters are run in parallell which is one of the reasons `golangci-lint` can perform so fast as it does. Running the linters sequentially would result in a major performance impact. Depending on the amount of issues reported even waiting for all results and sorting the output could result in a hit as well.

Is this only for your personal workflow or is there some underlying reason for this I'm missing?

**Comment by invidian** [comment]
>The linters are run in parallell which is one of the reasons golangci-lint can perform so fast as it does. Running the linters sequentially would result in a major performance impact. Depending on the amount of issues reported even waiting for all results and sorting the output could result in a hit as well.

Right, I don't mind running in parallel, but this should be separate from the printing part I would imagine. Can't warnings be sorted before being printed? This behavior could be hidden by a feature flag initially or if it would have some kind of performance penalty.

>Is this only for your personal workflow or is there some underlying reason for this I'm missing?

This is for the personal workflow. But I think whenever it's possible, it's good to have a deterministic output from the executed commands.

**Comment by iwankgb** [comment]
I guess that sorting is doable. Perhaps not by default (it might be an issue for large amount of violations) but by enabling sorting processor. @invidian would it be good-enough-fix for you?

**Comment by invidian** [comment]
Yes, if this could be configured in `.golangci.yml` file, it would be awesome.

BTW, I usually run with the following configuration, which I think amplifies the need for sorting.
```yaml
issues:
  max-same-issues: 0
  max-issues-per-linter: 0
```

If you have some hints how this could be implemented, I can have a look into it myself. I started browsing the code but passed for now not being able to follow the CLI logic.

**Comment by iwankgb** [comment]
I think I would start with, ekhm, [sort_processor.go](https://github.com/golangci/golangci-lint/blob/master/pkg/result/processors/sort_results.go) (I wasn't aware of it to be honest):
```yaml
output:
  sort-results: true
```

**Comment by iwankgb** [comment]
The flag seems to be missing from example configuration file and the docs, I will get it added :)

**Comment by invidian** [comment]
@iwankgb awesome, it seems this config works as expected :ok_hand:

**Comment by bombsimon** [comment]
Wow! I actually remember this now when you link it but I totally forgot about it. 💯
